src/main/java/com/mongodb/jdbc/MongoSQLColumnInfo.java

```diff
@@ -0,0 +1,171 @@
+package com.mongodb.jdbc;
+
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+import org.bson.BsonType;
+
+public class MongoSQLColumnInfo implements MongoColumnInfo {
+    private String datasource;
+    private String field;
+    private String bsonTypeName;
+    private BsonType bsonType;
+    private int jdbcType;
+    private boolean isPolymorphic;
+    private int nullable;
+
+    MongoSQLColumnInfo(String datasource, String field, MongoJsonSchema schema, int nullable)
+            throws SQLException {
+        this.datasource = datasource;
+        this.field = field;
+        // All schemata except Any and AnyOf must have a bsonType.
+        if (schema.bsonType != null) {
+            bsonTypeName = schema.bsonType;
+            bsonType = MongoColumnInfo.getBsonTypeHelper(schema.bsonType);
+            jdbcType = getJDBCTypeForBsonType(this.bsonType);
+            isPolymorphic = false;
+            this.nullable = nullable;
```

Comment:
I think the argument to this function should be something like `boolean fieldRequired` instead -- IIUC from the implementation, it doesn't actually represent nullability in the same way that the `nullable` member does, since nullability is not just dependent on the field's required status, but also on the actual value (or, more specifically, the possible values allowed by the schema)